src/runtime/HalideRuntime.h

```diff
@@ -110,10 +110,20 @@ extern void halide_mutex_lock(struct halide_mutex *mutex);
 extern void halide_mutex_unlock(struct halide_mutex *mutex);
 extern void halide_cond_signal(struct halide_cond *cond);
 extern void halide_cond_broadcast(struct halide_cond *cond);
-extern void halide_cond_signal(struct halide_cond *cond);
 extern void halide_cond_wait(struct halide_cond *cond, struct halide_mutex *mutex);
 //@}
 
+/** Functions for constructing/destroying/locking/unlocking arrays of mutexes. */
+struct halide_mutex_array {
+    struct halide_mutex *array;
```

Comment:
Does this struct need to be defined here? Would be nice if it were opaque to make the data type abstract.

i.e.
```struct halide_mutex_array;```

and then put the actual definition alongside the implementation of the methods below. This would let us change the implementation of the mutex array depending on the target.

Reply:
Done. Ran into some stupid C/C++ error when I defined an empty struct in fake pool thread.